CHANGELOG.md

```diff
@@ -1,3 +1,64 @@
+# 17.0.0
+
+_released `2025-05-07`_
+
+Version `17.0.0` is a breaking change release that contains several important upgrades and changes. Here is a brief breakdown of the main breaking changes (**:exclamation: denotes a breaking change**):
+
+1. ES Module support: It's now possible to build integrations using modern ESM syntax. This means we have changed the way we build integrations, and changed several dependencies.
+2. Schema changes: We have split authentication fields, input fields, and output fields into their own respective schemas, to allow for stricter schema checks and to prepare for future platform updates. Along with this, we have removed irrelevant or incompatible field types and properties from certain schemas (see the `schema` section below for more detail).
+3. No longer replace `{{curly brackets}}` outside of shorthand requests
+
+Here are the detailed release notes for this release (**note that ❗ denotes a breaking change**):
+
+### cli
+
+- :nail_care: ESM support added to `zapier init` command via the `--module` flag - supports Minimal and Typescript templates ([#976](https://github.com/zapier/zapier-platform/pull/976))
+- :hammer: `zapier build` now uses `esbuild` instead of `browserify` to detect dependencies ([#946](https://github.com/zapier/zapier-platform/pull/946))
+- :exclamation: `zapier build` and other local command updates: We have changed the way app definitions are imported to support integrations using ESM.
+- :hammer: Update `gulp-prettier` dependency from 4.0.0 to 5.0.0
+
+### core
+
+- :bug: HTTP 500 along with status codes >500 are caught in RPC client ([#974](https://github.com/zapier/zapier-platform/pull/974))
+- :hammer: Remove Bluebird library, replace with native promises ([#980](https://github.com/zapier/zapier-platform/pull/980))
+  - Context for integration logs is now tracked with a `whatHappened` property on the `_zapier` object
+- :hammer: Refactor middleware and lambda handler logic to use async/await instead of Promise chaining ([#980](https://github.com/zapier/zapier-platform/pull/980))
+- :exclamation: ESM Support: Two versions of `zapierwrapper.js`, one CJS and one ESM, loaded dynamically depending on the app type ([#965](https://github.com/zapier/zapier-platform/pull/965))
+  - Note: Because of this, you must use `zapier-platform-cli` version 17 to run `zapier` commands on an integration using `zapier-platform-core` version 17.
```

Comment:
CLI v17 _should_ work with Core v16. I remember I spent time making it work. Let me know if there's scenario that it doesn't work.

Reply:
Yes it generally works! I remembered there was one scenario that would not work and had a TODO to look up the details. It doesn't work if they are running CLI v17 and Core v16 **and specifically trying to use ESM** - you added [this helpful error](https://github.com/zapier/zapier-platform/blob/82b11aef29a4e7cb576431dba24cba0066c5057b/packages/cli/src/utils/zapierwrapper.js#L27-L32) that will come up. Since we have a descriptive error and ESM is explicitly supported starting with Core v17, maybe it's not necessary to include in the changelog - I can remove it.